docs/local-build/mine-validate.md

```diff
@@ -1,125 +1,337 @@
+ ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+                                    1                   287.57 β       0.0000       0.0000       0.0000       0.0000       0.0000    ρ38841066       0.0000
+
+                                                                       Wallet free balance: 99,994.3638 τ
 ```
-Transfer funds from the Alice account to cover it and try again. Consult `btcli w list` and carefully check the ss58 address of the destination coldkey (in this case, the one with the name `validator`).
 
-```shell
-btcli wallet transfer \
---amount 11 \
---wallet.name alice \
---destination "5EEy34..." \
---network ws://127.0.0.1:9945
+</details>
+
+If the validator wallet has a validator permit, an asterisk (`*`) is shown under the `VPERMIT` column for the corresponding subnet in the response table.
+
+</TabItem>
+<TabItem value="python SDK" label="Using Bittensor SDK">
+Input the following lines in your Python environment, replacing `NETUID`, `WALLET_NAME`, and `WALLET_HOTKEY` with the target subnet ID, the name of the wallet, and the associated hotkey, respectively.
+
+```python
+import bittensor as bt
+network=bt.subtensor(network="local")
+subnet = network.metagraph(NETUID)
+wallet = bt.wallet( name = 'WALLET_NAME', hotkey = 'HOTKEY' )
+my_uid = subnet.hotkeys.index( wallet.hotkey.ss58_address )
+print(f'Validator permit: {subnet.validator_permit[my_uid]}')
+```
+
+The command outputs `True` or `False` depending on whether the validator hotkey has a permit.
+</TabItem>
+</Tabs>
+
+:::info Validator Permits on Localnet Subnets
+On localnet subnets, competition for permits is typically minimal. After staking TAO to the validator hotkey, the neuron should become eligible for a validator permit. If it does not yet have one, wait until the end of the subnet’s tempo.
+:::
+
+## 3. Pull the `subnet-template` repo
+
+The `subnet-template` repo contains the core logic for the subnet miner and validator. It features a simple `dummy` protocol where miners multiply input values by 2, while validators evaluate responses and update network weights based on performance.
+
+To begin, clone the subnet-template GitHub repository and navigate into its directory:
+
+```sh
+git clone https://github.com/opentensor/subnet-template.git
+cd subnet-template
+```
+
+## 4. Run the miner and validator
+
+After getting the validator permits, you can now run the validator alongside the miner to begin participating in the subnet.
+
+Begin by starting the miner process to produce and submit work to the subnet. Then, run the validator process to evaluate miner outputs and set weights for the network.
+
+### Start the miner process
+
+To start the miner, run the following Python script in the `subnet-template` directory:
+
+```sh
+python miner.py \
```

Comment:
Just realized you should probably add `--logging.info debug` to the miner.py and validator.py commands, it makes them output a lot more useful info